data/mods/gen3/abilities.js

```diff
@@ -143,8 +143,8 @@ let BattleAbilities = {
 		inherit: true,
 		onStart(pokemon) {
 			this.add('split');
-			for (const line of [false, this.sides[0], this.sides[1], true]) {
-				if (line === true || line === pokemon.side) {
+			for (const line of [false, this.p1, this.p2, true]) {
+				if (line === true || line && line.n === pokemon.side.n % 2) {
```

Comment:
It looks like these kinds of functions should pass `0` and `1` in side of `p1` and `p2`

Reply:
I don’t understand.

Reply:
I think he means 'instead'? like `[false, 0, 1, true]`?

Reply:
That actually ended up fixing showing exact HP for players 3 and 4.

Reply:
Yes I did mean "instead".